### PR TITLE
[SIG-2706] replaces the SelectInput in incident-management with an ASC component

### PR DESCRIPTION
**Pull request by aditudorache**
This PR replaces the SelectInput in incident-management with an ASC component. The onChange method of this component is not required anymore in order to address the uncontroled component scenarios,

### CODE REVIEW

File: src/signals/incident-management/components/SelectInput/index.js [hunk 1]
```diff
@@ -1,53 +1,44 @@
...
+  const options = values.map(({ key, value, slug }) => ({
+    key: useSlug ? slug : key,
+    name: key ? value : emptyOptionText || value,
+    value: useSlug ? slug : key,
+  }));
+
+  const render = ({ handler }) => {
+    const value = handler()?.value;
+    const keyName = useSlug ? 'slug' : 'key';
+    const { [keyName]: info } = (value && values.find(p => p && p[keyName] === value)) || {};
```
Comment (janjaap) [inline src/signals/incident-management/components/SelectInput/index.js:30]: This is not correct; the prop you should be using from the subcategory's data is `description`. Hardly any subcategory has a description. You can add one yourself to test it.

File: src/signals/incident-management/components/SelectInput/index.js [hunk 2]
```diff
@@ -1,53 +1,44 @@
...
+
+    return (
+      <Wrapper>
+        {display && <Label htmlFor={`form${name}`}>{display}</Label>}
+
+        <SelectInputComponent name={name} data-testid={name} id={`form${name}`} {...handler()} options={options} />
```
Comment (janjaap) [inline src/signals/incident-management/components/SelectInput/index.js:36]: I've checked out the branch and the design seems to have a height of 44 pixels. The standard height, though, is 40 pixels. Can you ask UX what it should be?

File: src/signals/incident-management/components/SelectInput/index.js (hunk 1 above)
Comment (aditudorache) [reply src/signals/incident-management/components/SelectInput/index.js:30]: ok, i have added it.

File: src/signals/incident-management/components/SelectInput/index.js (hunk 2 above)
Comment (aditudorache) [reply src/signals/incident-management/components/SelectInput/index.js:36]: UX says 40 is OK

File: src/components/SelectInput/index.js [hunk 3]
```diff
@@ -11,9 +11,13 @@ const SelectInput = ({ onChange, name, options, value }) => (
   </Select>
 );
 
+SelectInput.defaultValues = {
```
Comment (janjaap) [inline src/components/SelectInput/index.js:14]: `defaultProps`?

File: src/components/SelectInput/index.js [hunk 4]
```diff
@@ -11,9 +11,13 @@ const SelectInput = ({ onChange, name, options, value }) => (
...
+  onChange: () => { },
```
Comment (janjaap) [inline src/components/SelectInput/index.js:15]: Didn't the linter warn about the space between the braces?

File: src/components/SelectInput/index.js (hunk 3 above)
Comment (aditudorache) [reply src/components/SelectInput/index.js]: right!

File: src/components/SelectInput/index.js (hunk 4 above)
Comment (aditudorache) [reply src/components/SelectInput/index.js]: not, strange.

File: src/signals/incident-management/components/SelectInput/index.js [hunk 5]
```diff
@@ -1,53 +1,44 @@
...
+    const keyName = useSlug ? 'slug' : 'key';
+    const { _display: description } = (value && values.find(p => p && p[keyName] === value)) || {};
+
+    return (
+      <Wrapper>
+        {display && <Label htmlFor={`form${name}`}>{display}</Label>}
+
+        <SelectInputComponent name={name} data-testid={name} id={`form${name}`} {...handler()} options={options}/>
+
+        {description && <Info>{description}</Info>}
```
Comment (janjaap) [inline src/signals/incident-management/components/SelectInput/index.js:38]: Just approved the PR, but then realised that this change shouldn't be part of the PR.

Comment (aditudorache) [reply src/signals/incident-management/components/SelectInput/index.js:38]: I'll remove it!
